src/renderer/screens/manager/DeviceStorage/index.js

```diff
@@ -255,4 +266,4 @@ const DeviceStorage = ({ state, deviceInfo }: *) => {
   );
 };
 
-export default DeviceStorage;
+export default memo<Props>(DeviceStorage, isEqual);
```

Comment:
I don't think it's a good idea to use the (deep) isEqual of lodash here. especially that it's going to apply each time on the state object! which is often changing anyway, we should probably instead split the component in more pieces

Reply:
This one was unnecessary sorry :bowing_man: 
I replaced the deep Equal with a more tailored solution on each item lines that's where the re-render bottleneck is concentrated when progress is ongoing :+1: